app/src/main/java/com/x8bit/bitwarden/data/platform/util/StringExtensions.kt

```diff
@@ -71,7 +71,18 @@ fun String.hasPort(): Boolean {
  * Extract the host from this [String] if possible, otherwise return null.
  */
 @OmitFromCoverage
-fun String.getHostOrNull(): String? = this.toUriOrNull()?.host
+fun String.getHostOrNull(): String? = this.toUriOrNull()
+    ?.let { uri ->
+        if (
+            uri.host == null &&
+            this.contains(".") &&
+            !this.hasHttpProtocol()
```

Comment:
Can we add a little comment or doc just explaining why these extra steps are needed